docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,187 @@
+# Storage node installation and update on Linux
+
+## Abstract
+
+The design document outlines the steps needed to ease installation and provide auto-update of storage nodes on Linux without the use of Docker.
+
+
+## Background
+
+
+The idea is to have two main components, like for Windows: the storage node binary and a storage node updater that will take care of updating the storage node binary according to the rollout version mechanism.
+
+The parts we need:
+- storagenode as a service
+- a system for updating the storagenode binary aka. the updater (with rollout versioning support)
+- a system for updating the updater 
+- a way to collect the configuration data from the user during the installation
+- packaging to ship the above
+
+Linux systems are not unified as windows is. Hence, we have to make choices that will allow us to cover most linux users, with special regards to the raspbian system.
+
+Most commonly used linux distributions seem to be (there are no reliable stats):
+- debian-based (Debian, Ubuntu, Mint, Kali, Raspbian)
+- Red Hat-based (Fedora)
+- Arch Linux (Manjaro is on the rise according to [distrowatch](https://distrowatch.com/dwres.php?resource=popularity)).
+
+All these distributions are shipped with systemd as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised storage
+  - Identity directory 
+  - Storage directory
+- Generate `config.yaml` file with the user configuration.
+
+The default value for these directories can be defined using the [XDG Base Directory](https://wiki.archlinux.org/index.php/XDG_Base_Directory).
+
+We choose to reuse the storagenode-updater and the recovery mechanism used in windows. They will be daemonized using systemd. The storagenode updater will auto-update. A recovery will be triggered if the updated updater service fails to restart.
+We will use debconf to retrieve user data. 
```

Comment:
Do we intend for the deb package to be installable on redhat or Arch-based platforms, by way of the 'alien' tool? Because the debconf part will not be portable and will fail when users try that.

Secondly, when do we plan to use debconf to retrieve user data? Only on install, so that we can generate an appropriate config file? Or do we plan to query debconf every time we start the service? (If the latter case, that might raise some problems.)

Reply:
I guess the alien part of the question is answered already- forgot to come back and take it out once I finished the review :)

Reply:
Personally I was thinking of querying debconf, so that the user can change their settings when they want using a standardized method. I know this would need adaptation for other package formats. But I'm not sure that's the best approach.

Reply:
The standardized method on Debian would be something like "`dpkg-reconfigure storj-storagenode`"- which would rerun our configuration script, so we can still generate or regenerate a config file from debconf values. It's considered bad practice to query debconf directly from an application.

Reply:
Ok, I wasn't aware of this, thank you :) So now I wonder, does it make any sense to use debconf?

Reply:
The benefits of using debconf would be (a) it's a much more flexible way to ask the admin for configuration settings; the various frontends for debconf work in GUI environments, web consoles, smart terminals, dumb terminals, braille displays, etc. And (b) debconf settings can also be pre-seeded: the user can specify what they want the settings to be before even installing the software, so that the installation can be automated and unattended. Also (c) is that it's very translator-friendly, for when we want to add localization.

It does add some complexity, though, certainly, so it would probably be ok for us to skip debconf altogether for now and put it in if/when we need it.

Reply:
As it is already integrated in the demo debian package, I guess we'll keep debconf then.